File: MoonReader/Models/Note.swift
//
//  Note.swift
//  MoonReader
//
//  Note/Annotation model
//

import Foundation

struct Note: Identifiable, Codable {
    var id: UUID
    var bookId: UUID
    var chapter: Int
    var position: Double
    var text: String
    var createdAt: Date
    var modifiedAt: Date
    
    init(
        id: UUID = UUID(),
        bookId: UUID,
        chapter: Int = 0,
        position: Double = 0.0,
        text: String,
        createdAt: Date = Date(),
        modifiedAt: Date = Date()
    ) {
        self.id = id
        self.bookId = bookId
        self.chapter = chapter
        self.position = position
        self.text = text
        self.createdAt = createdAt
        self.modifiedAt = modifiedAt
    }
}

File: MoonReaderRN/src/models/Note.ts
import { Model } from '@nozbe/watermelondb';
import { field, text, date } from '@nozbe/watermelondb/decorators';

// TypeScript interface
export interface Note {
  id: string;
  bookId: string;
  chapter: number;
  position: number;
  text: string;
  createdAt: Date;
  modifiedAt: Date;
}

// WatermelonDB Model class
export class NoteModel extends Model {
  static table = 'notes';

  @text('bookId') bookId!: string;
  @field('chapter') chapter!: number;
  @field('position') position!: number;
  @text('text') text!: string;
  @date('createdAt') createdAt!: Date;
  @date('modifiedAt') modifiedAt!: Date;
}

